### DISCUSSION TITLE
nix user repo and order of pipelines

### DISCUSSION THREAD

**Discussion by till** [goreleaser/goreleaser#4479] [Q&A] [answered]
Hi,

we wanted to create a nix user repository to publish a cli tool (code is private for now). So far, our release workflow looks like this:

```yaml
# pseudo workflow
job1
  steps:
  - goreleaser release [all bunch of targets (docker, homebrew, nix, ...) 
  - action/upload_artifacts
 
job2:
  needs: job1
  steps:
  - download artifacts
  - upload to s3 bucket, etc.
 ```

I noticed that the nix build tries to "validate" a pre fetch thing (I am a nix-noob still). And that URL points to our s3, which is not available yet (error 404) as the artifacts are uploaded in the other `job2`. Therefor, publishing/updating of the nur fails the release job (it doesn't skip, it actually fails the job).

Long story short, I saw there are `blobs` now and custom publishers. What is the order of these things? So for example, when I upload a bunch of artifacts using a blob (to s3), will this be available before the nur is build?

Thank you!

**Comment by caarlos0** [comment, marked as answer]
hey, in theory if you use the `blobs` feature to upload to s3, it should all work out

custom publishers I think will work too.

**Reply by till** [reply]
I just looked into it a bit. It seems like we cannot use `blobs` because we can't set the object ACL (`public-read`) there. The only option seems to be a custom publisher then?

**Reply by caarlos0** [reply]
ah yes, for now it is... this is something I want to fix/improve when I get to work on #4407

**Reply by till** [reply]
I can report, it works with a custom publisher. Now all that's left is figuring out what this nix thing is all about. 😁
